NDTensors/ext/NDTensorCUDA/adapt.jl

```diff
@@ -0,0 +1,30 @@
+#
+# Used to adapt `EmptyStorage` types
+#
+
+struct NDTensorCuArrayAdaptor{B} end
+## TODO make this work for unified. This works but overwrites CUDA's adapt_storage. This fails for emptystorage...
+@inline function NDTensors.cu(xs; unified::Bool=false)
+  return fmap(
+    x -> adapt(NDTensorCuArrayAdaptor{unified ? Mem.UnifiedBuffer : Mem.DeviceBuffer}(), x),
+    xs,
+  )
+end
+
+function Adapt.adapt_storage(
+  ::NDTensorCuArrayAdaptor{B}, xs::AbstractArray{T,N}
+) where {T,N,B}
+  return isbits(xs) ? xs : CuArray{T,1,B}(xs)
+end
+
+function NDTensors.adapt_storagetype(
+  ::NDTensorCuArrayAdaptor{B}, xs::Type{EmptyStorage{ElT,StoreT}}
+) where {ElT,StoreT,B}
```

Comment:
Same comment as above about trying to avoid `where` as much as possible.

Reply:
Maybe you can use the `storagetype` function you defined below for `EmptyStorage`.